pkg/apis/serving/v1alpha2/explainer.go

```
@@ -33,24 +32,41 @@ const (
 )
 
 // Returns a URI to the explainer. This URI is passed to the model-initializer via the ModelInitializerSourceUriInternalAnnotationKey
-func (m *ExplainerSpec) GetStorageUri() string {
-	return getExplainerHandler(m).GetStorageUri()
+func (e *ExplainerSpec) GetStorageUri() string {
+	explainer, err := getExplainer(e)
+	if err != nil {
+		return ""
+	}
+	return explainer.GetStorageUri()
 }
 
-func (m *ExplainerSpec) CreateExplainerContainer(modelName string, predictorHost string, config *ExplainersConfig) *v1.Container {
-	return getExplainerHandler(m).CreateExplainerContainer(modelName, predictorHost, config)
+func (e *ExplainerSpec) CreateExplainerContainer(modelName string, predictorHost string, config *ExplainersConfig) *v1.Container {
+	explainer, err := getExplainer(e)
+	if err != nil {
```

Also swallowing error here